### ISSUE TITLE
`column source_...~1.id does not exist` when migration adds a link property to existing link on new database

### ISSUE THREAD

**Issue by jackfischer** [edgedb/edgedb#5060]
- EdgeDB Version: 2.10+66168b2
- EdgeDB CLI Version: 2.2.6+7eabbf9
- OS Version: macOS 13.1

Steps to Reproduce:

1. Create a fresh database via `edgedb project init`
2. Create two objects with a link property between them
3. Migrate to add a link property to the link

Note that this does not occur when migrating an existing database with data in it

Example migration

```
CREATE MIGRATION m1e5lg4zthhiwftctkhu42zk3an4wlckofxbbacknb6mvqucswr6tq
    ONTO m145ubndj6w5mhlvoea2qdomtgvh6ziia43mvbwsuwu43d6h3nq4ua
{
  ALTER TYPE default::Survey {
      ALTER LINK recipients {
          CREATE PROPERTY destination -> std::str {
              SET default := 'email';
          };
      };
  };
};
```

```
$ edgedb migrate
edgedb error: InternalServerError: column source_23dcd9ce-af36-11ed-b05e-69076c83b0e2~1.id does not exist
  Hint: This is most likely a bug in EdgeDB. Please consider opening an issue ticket at https://github.com/edgedb/edgedb/issues/new?template=bug_report.md
  Server traceback:
      edb.errors.InternalServerError: column source_23dcd9ce-af36-11ed-b05e-69076c83b0e2~1.id does not exist
```

Schema:

```
module default {
    type Survey {
        multi link recipients -> User;
    }
    type User{}
}
```

```
module default {
    type Survey {
        multi link recipients -> User {
            property destination -> str {
                default := "email";
            }
        }
    }
    type User{}
}
```

**Comment by msullivan** [comment]
> Note that this does not occur when migrating an existing database with data in it
I think this is a red herring; I was able to reproduce it with data in the database.

The `default` on the link property is key. The part where things fail is in trying to apply the default property to potentially existing links.

**Comment by jackfischer** [comment]
Interesting. Noting for posterity from slack that "Note that this does not occur when migrating an existing database with data in it" is probably wrong an an artifact of the migration previously running successfully on <2.10

**Comment by msullivan** [comment]
But, interestingly, despite that, adding a default to an existing linkprop works

**Comment by jackfischer** [comment]
Smells possibly related to this fix introduced in 2.10, if only for the link prop connection?
https://github.com/edgedb/edgedb/issues/4961
